### PR TITLE
NN-3463:

### PR DESCRIPTION
**Pull request by JamieDraperUK**
- create initial tables to store restricted patients
- update circlci to run tests against postgres
- run tests locally against postgres
- enable JPA audit

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsrestrictedpatientsapi/services/RestrictedPatientsService.kt [hunk 1]
```diff
@@ -3,34 +3,48 @@ package uk.gov.justice.digital.hmpps.hmppsrestrictedpatientsapi.services
...
 @Service
 class RestrictedPatientsService(
   private val prisonApiGateway: PrisonApiGateway,
-  private val prisonerSearchApiGateway: PrisonerSearchApiGateway
+  private val prisonerSearchApiGateway: PrisonerSearchApiGateway,
+  private val restrictedPatientsRepository: RestrictedPatientsRepository
 ) {
+
+  @Transactional
   fun dischargeToHospital(dischargeToHospital: DischargeToHospitalRequest): DischargeToHospitalResponse {
```
Comment (JamieDraperUK) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsrestrictedpatientsapi/services/RestrictedPatientsService.kt:23]: Telemetry events coming in later PRs

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsrestrictedpatientsapi/model/entities/RestrictivePatient.kt [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import java.time.LocalDateTime
+import javax.persistence.Column
+import javax.persistence.Entity
+import javax.persistence.EntityListeners
+import javax.persistence.Table
+
+@Entity
+@Table(name = "RESTRICTED_PATIENTS")
+@EntityListeners(AuditingEntityListener::class)
+class RestrictivePatient(
```
Comment (sp-watson) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsrestrictedpatientsapi/model/entities/RestrictivePatient.kt:15]: Just checking - is that name correct - it doesn't match the table name?!

Comment (JamieDraperUK) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsrestrictedpatientsapi/model/entities/RestrictivePatient.kt:15]: Ah, no it’s a typo. Good spot.

File: docker-compose-test.yml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+version: '3.1'
```
Comment (sp-watson) [inline docker-compose-test.yml:1]: I assume this needs to be executed when running int tests - if so, it should be part of the README

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsrestrictedpatientsapi/model/entities/RestrictivePatient.kt (hunk 2 above)
Comment (sp-watson) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsrestrictedpatientsapi/model/entities/RestrictivePatient.kt:15]: Just checking - is that name correct - it doesn't match the table name?!

**Review by sp-watson — APPROVED** [review]

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsrestrictedpatientsapi/model/entities/RestrictivePatient.kt (hunk 2 above)
Comment (JamieDraperUK) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsrestrictedpatientsapi/model/entities/RestrictivePatient.kt:15]: Good spot will fix in the next PR

**Pull request merged**